### PR TITLE
Bug 1451261, complete change to svg icon system, remove FontAwesome lib

### PR DESCRIPTION
**Pull request by schalkneethling**
Ok, phew. I think this does it. With these changes, FontAwesome (the web font) is not used anywhere except by CKEditor.

I used the following pages to test:

* https://developer.mozilla.org/en-US/docs/Sandbox/Indicators
* https://developer.mozilla.org/en-US/docs/Sandbox/All-Indicators

You can create local copies of these using:

```
make bash
...
./manage.py scrape_document https://developer.mozilla.org/en-US/docs/Sandbox/All-Indicators
```

I also did a bit of a browse around some of the documentation pages, the landing page etc. All looks to be working as expected. This is rather big and will probably need some time for testing. Please let me know should you find anything that looks to be broken, or not as expected.

Once this is merged, I believe it will take care of the following issues:

* https://app.zenhub.com/workspace/o/mdn/sprints/issues/407
* https://app.zenhub.com/workspace/o/mdn/sprints/issues/408
* https://app.zenhub.com/workspace/o/mdn/sprints/issues/409

There is more cleanup that can be done, but I will leave that for a separate PR. Thank you in advance for your reviews.

### CODE REVIEW

**Comment by schalkneethling** [comment]
I will resolve the above conflicts and update the PR soon.

**Comment by schalkneethling** [comment]
This also resolves related bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1502813

File: kuma/static/styles/components/wiki/indicators.scss [hunk 1]
```diff
@@ -259,61 +261,87 @@ icons for both types of indicators
...
+    @include indicator-svg-icon($path-to-svg-file-icons + 'file.svg');
 
     &:before {
-        @include vendorize(transform, scaleY(-.8) scaleX(.8));
+        @include vendorize(transform, scaleY(-1) scaleX(1));
     }
 }
 
 .obsolete {
+<<<<<<< HEAD
```
Comment (jwhitlock) [inline kuma/static/styles/components/wiki/indicators.scss]: This merge conflict wasn't resolved, is breaking the build

Comment (schalkneethling) [reply kuma/static/styles/components/wiki/indicators.scss]: Meh, did not see that one. Will fix it up. 😞

Comment (schalkneethling) [reply kuma/static/styles/components/wiki/indicators.scss]: Fixed up.

**Comment by jwhitlock** [comment]
I'll let @hobinjk see if the CSS looks reasonable.  Looking at the test pages, there are some differences, which might be minor or may be unintended. 

@schalkneethling can you say if these are intentional changes or bugs? For the screenshots, the right side is production and Font Awesome (FA), the left is development and SVG.  

In this screenshot, from https://developer.mozilla.org/en-US/docs/Sandbox/All-Indicators:

* The FontAwesome banners are taller, probably because the icons are larger.
* The FA thumb icon is an outline with a solid sleeve, while the SVG thumb is filled-in with a solid sleeve
* The FA note icon is an outline, versus a solid icon for SVG
* The FA translationInProgress is a planet, while the SVG icon is a pencil

screenshot 2018-11-02 13 46 30

Further down in examples, the FA example has a red frowny face icon and a green smiley icon in the lower right, and the SVG version doesn't:

screenshot 2018-11-02 13 50 40

The "external link" icon is blue in FA, and black in SVG:

screenshot 2018-11-02 13 52 10

``non-standard_inline`` (great name 🙄) is yellow in FA, black in SVG:

screenshot 2018-11-02 13 53 04

``obsolete_inline`` is red in FA, black in SVG:

screenshot 2018-11-02 13 54 34

``experimental_inline`` is blue in FA, black in SVG:

screenshot 2018-11-02 13 55 10

System Messages have clock and frowny face icons in FA, and no icons in SVG:

screenshot 2018-11-02 13 55 57

The redirect icon is yellow-brown in FA, black in SVG:

screenshot 2018-11-02 13 56 47

Moving on to https://developer.mozilla.org/en-US/docs/Sandbox/Indicators, there's a single difference. In FA, the "Mailing list/Newsgroup" and "RSS feed" buttons have external link icons, and they are gone in SVG:

screenshot 2018-11-02 13 59 08

**Comment by schalkneethling** [comment]
> System Messages have clock and frowny face icons in FA, and no icons in SVG:
> Moving on to developer.mozilla.org/en-US/docs/Sandbox/Indicators, there's a single
> difference. In FA, the "Mailing list/Newsgroup" and "RSS feed" buttons have 
> external link icons, and they are gone in SVG:

> Not intentional. This worked prior to open the PR, will have to see what it is missing.

Turns out Firefox only supports `base64` encoded strings.

**Comment by schalkneethling** [comment]
I now remember the reason `non-standard_inline` and `obsolete_inline` is black. The macro does not add anything to "highlight" it as being standalone and inline. Therefore I cannot just set `.icon-trash` to be red, as all instances of that class will then be red.

We therefore have two choices:

1. Leave it black
2. Update the relevant macros to add a class such as `standalone-inline` or `inline` so that I can treat these differently.

Thoughts @jwhitlock?

**Comment by jwhitlock** [comment]
We had a chance to talk about this over, and I think we'd like to get the KumaScript macro changes merged first or at least in the same deploy, so we don't lose colors on some inline indicators. @schalkneethling is this what you remember as well?

* [ ] [non-standard_inline](https://developer.mozilla.org/en-US/docs/Sandbox/All-Indicators#General) - Update [NonStandardBadge](https://github.com/mdn/kumascript/blob/master/macros/NonStandardBadge.ejs#L47) to add a class to allow coloring yellow in inline mode.

   screenshot 2018-11-02 13 53 04

* [ ] [obsolete_inline](https://developer.mozilla.org/en-US/docs/Sandbox/All-Indicators#Obsolete_inline) - Update [ObsoleteBadge](https://github.com/mdn/kumascript/blob/778a730beaf5e6bc3644a92e788c0458e8b50a18/macros/ObsoleteBadge.ejs#L46) to add a class to allow coloring red in inline mode (and fix the extra quotes in the macro)

   screenshot 2018-11-02 13 54 34

* [ ] [System Error Boxes](https://developer.mozilla.org/en-US/docs/Sandbox/All-Indicators#Well_crap) - Make the clock and frowny face visible.

   screenshot 2018-11-02 13 55 57

* [ ] [TopicBox](https://developer.mozilla.org/en-US/docs/Sandbox/Indicators) - Update [DiscussionList](https://github.com/mdn/kumascript/blob/master/macros/DiscussionList.ejs#L183-L192) or other code to make the external icons visible.

   screenshot 2018-11-02 13 59 08

Other items we're changing and see if anyone complains:

* Banner heights, larger icons
* Icon change for the thumb and note icons
* ``translationInProgress`` changes from a planet to a pencil
* Lose smile and frown from ``example-good`` and ``example-bad`` code samples
* External link icons turn from blue to black
* Experimental inline beaker goes from blue to black
* Redirect icon goes from brown to black

And, could still use 👀  and a 👍 from @hobinjk.

**Comment by schalkneethling** [comment]
> Lose smile and frown from example-good and example-bad code samples

We will not loose these, the fix for that will be in the coming update of the PR. Thanks @jwhitlock

**Comment by schalkneethling** [comment]
@jwhitlock Updated, and also opened relevant PR on the KumaScript side:
https://github.com/mdn/kumascript/pull/981

**Comment by jwhitlock** [comment]
With the current branch plus https://github.com/mdn/kumascript/pull/981 (for screenshots, left is this PR / SVG, right is production / Font Awesome, standard DPI screen):

* [x] [non-standard_inline](https://developer.mozilla.org/en-US/docs/Sandbox/All-Indicators#General) - yellow color matches

screenshot 20
![screenshot 2018-11-06 09 25 08
18-11-06 09 00 00](https://user-images.githubusercontent.com/286017/48073719-f66c9f00-e1a4-11e8-914c-77e7619f6099.png)

* [x] [obsolete_inline](https://developer.mozilla.org/en-US/docs/Sandbox/All-Indicators#Obsolete_inline) - red color matches

screenshot 2018-11-06 09 20 20

* [x] [System Error Boxes](https://developer.mozilla.org/en-US/docs/Sandbox/All-Indicators#Well_crap) - Clock and frowny faces are now visible

screenshot 2018-11-06 09 25 08

* [x] [TopicBox](https://developer.mozilla.org/en-US/docs/Sandbox/Indicators) - External icons are back, and a bit sharper in the PR than in production.

screenshot 2018-11-06 09 36 37

* [x] ``example-good`` and ``example-bad`` code samples - Green smile and red frown in lower right corners

screenshot 2018-11-06 09 38 03

I've also looked at them on a high DPI display, it looks similar.

These are the changing items:

* Banner heights, larger icons
* Icon change for the thumb and note icons
* ``translationInProgress`` changes from a planet to a pencil
* External link icons turn from blue to black
* Experimental inline beaker goes from blue to black
* Redirect icon goes from brown to black

File: jinja2/includes/icons/arrows/caret-left.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="11" height="28"><path d="M10 7v14c0 .547-.453 1-1 1a.99.99 0 0 1-.703-.297l-7-7C1.11 14.516 1 14.265 1 14s.109-.516.297-.703l7-7A.996.996 0 0 1 9 6c.547 0 1 .453 1 1z"/></svg>
```
Comment (jwhitlock) [inline jinja2/includes/icons/arrows/caret-left.svg]: Nit: needs a newline at end of file

File: jinja2/includes/icons/arrows/caret-right.svg [hunk 3]
```diff
@@ -1,3 +1 @@
-<svg class="icon icon-caret-right" xmlns="http://www.w3.org/2000/svg" width="9" height="28" viewBox="0 0 9 28" aria-hidden="true">
-    <path d="M9 14a.99.99 0 0 1-.297.703l-7 7A.996.996 0 0 1 1 22c-.547 0-1-.453-1-1V7c0-.547.453-1 1-1a.99.99 0 0 1 .703.297l7 7A.996.996 0 0 1 9 14z"/>
-</svg>
+<svg xmlns="http://www.w3.org/2000/svg" width="19" height="28"><path d="M17.297 13.703L5.703 25.297a.99.99 0 0 1-1.406 0l-2.594-2.594a.99.99 0 0 1 0-1.406L10 13 1.703 4.703a.99.99 0 0 1 0-1.406L4.297.703a.99.99 0 0 1 1.406 0l11.594 11.594a.99.99 0 0 1 0 1.406z"/></svg>
```
Comment (jwhitlock) [inline jinja2/includes/icons/arrows/caret-right.svg]: Nit: needs a newline at end of file

File: jinja2/includes/icons/arrows/chevron-left.svg [hunk 4]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="21" height="28"><path d="M18.297 4.703L10 13l8.297 8.297a.99.99 0 0 1 0 1.406l-2.594 2.594a.99.99 0 0 1-1.406 0L2.703 13.703a.99.99 0 0 1 0-1.406L14.297.703a.99.99 0 0 1 1.406 0l2.594 2.594a.99.99 0 0 1 0 1.406z"/></svg>
```
Comment (jwhitlock) [inline jinja2/includes/icons/arrows/chevron-left.svg]: nit: needs newline at end of file

File: kuma/static/styles/includes/_svg_data_uri_icons.scss [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+$icon-red-frown: 'data:image/svg+xml;base64,  PHN2ZyBzdHlsZT0iZmlsbDogI2U2NjQ2NTsiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgd2lkdGg9IjI0IiBoZWlnaHQ9IjI4Ij48cGF0aCBkPSJNMTcuNzE5IDE5LjIwM2EuOTkzLjk5MyAwIDAgMS0uNjQxIDEuMjUgMS4wMDcgMS4wMDcgMCAwIDEtMS4yNjYtLjY1NkMxNS4yOTYgMTguMTI1IDEzLjc0OSAxNyAxMS45OTkgMTdzLTMuMjk3IDEuMTI1LTMuODEzIDIuNzk3Yy0uMTcyLjUzMS0uNzM0LjgyOC0xLjI1LjY1Ni0uNTMxLS4xNzItLjgyOC0uNzE5LS42NTYtMS4yNUM3LjA2MSAxNi42ODcgOS4zNTggMTUgMTEuOTk5IDE1czQuOTM3IDEuNjg3IDUuNzE5IDQuMjAzek0xMCAxMGMwIDEuMTA5LS44OTEgMi0yIDJzLTItLjg5MS0yLTIgLjg5MS0yIDItMiAyIC44OTEgMiAyem04IDBjMCAxLjEwOS0uODkxIDItMiAycy0yLS44OTEtMi0yIC44OTEtMiAyLTIgMiAuODkxIDIgMnptNCA0YzAtNS41MTYtNC40ODQtMTAtMTAtMTBTMiA4LjQ4NCAyIDE0czQuNDg0IDEwIDEwIDEwIDEwLTQuNDg0IDEwLTEwem0yIDBjMCA2LjYyNS01LjM3NSAxMi0xMiAxMlMwIDIwLjYyNSAwIDE0IDUuMzc1IDIgMTIgMnMxMiA1LjM3NSAxMiAxMnoiLz48L3N2Zz4=';
```
Comment (hobinjk) [inline kuma/static/styles/includes/_svg_data_uri_icons.scss]: nit: inconsistent spacing (I think it's meant to align the PHNs but it falls apart pretty quickly)

**Comment by schalkneethling** [comment]
Thanks @jwhitlock and @hobinjk - Updated.

**Pull request merged**